### ISSUE TITLE
Failed to install chart -  Required value: must specify a volume type

### ISSUE THREAD

**Issue by defaultbr** [clemlesne/azure-pipelines-agent#8]
Hello, thanks for making the awesome chart.

I'm having throubles when applying the chart, the error is:

`
Error: UPGRADE FAILED: failed to create resource: Pod "azdo-agents-cluster-non-production-azure-pipelines-agent-3" is invalid: [spec.volumes[0]: Required value: must specify a volume type, spec.volumes[1]: Required value: must specify a volume type, spec.containers[0].volumeMounts[0].name: Not found: "azp-work", spec.containers[0].volumeMounts[1].name: Not found: "local-tmp"]
`

Helm command:

> helm upgrade 
>  azdo-agents-cluster-non-production stable/azure-pipelines-agent --version 4.16.0 
>  -f values.yml 
>  --kube-context=my-context
>  --namespace=azdo-agents
>  --install 
>  --wait 
>  --set serviceName=azdo-agents-cluster-non-production
>  --output table

Chart Version: 4.16.0

My values.yaml

```
pipelines:
  cacheSize: 10Gi
  cacheType: gp2

  tmpdirSize: 1Gi
  tmpdirType: gp2
    
  # capabilities: []
  url: https://dev.azure.com/....
  pat: ....
  pool: default-non-production  
```

`kubectl get storageclass`
```
NAME            PROVISIONER             RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION 
gp2 (default)   kubernetes.io/aws-ebs   Delete          WaitForFirstConsumer   false                  
```

**Comment by clemlesne** [comment]
Hello @defaultbr,

First thank you for your kind message. I do my best!

Relative to your issue:

The error says the Pod YAML schema is not right.

The Pod is deployed with ephemeral storage, [available from Kubernetes 1.23](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#generic-ephemeral-volumes). With this, a volume cannot technically be reused nor shared between Pods.

**Can you confirm your Kubernetes version?**

**Comment by defaultbr** [comment]
Jesus, you are right, he squad that handle the clusters told me that they updated the cluster version, but they don't told me that the version was 1.22 (not latest or stable), that's a shame.

Thank youu so much, I will create another cluster for my purpouse
